### DISCUSSION TITLE
How to add a list of coordinate to open3d

### DISCUSSION THREAD

**Discussion by assia855** [isl-org/Open3D#5152] [Q&A]
I have a list of coordinate:
`[[(0.0, 0.0), (1.0, 2.0), (14.0, 14.0)],
 [(0.0, 1.0), (3.0, 2.0), (38.0, 37.0)],
 [(0.0, 1.0), (3.0, 4.0), (38.0, 38.0)],
 [(0.0, 0.0), (23.0, 24.0), (10.0, 10.0)],
 [(0.0, 0.0), (24.0, 25.0), (10.0, 10.0)],
 [(0.0, 0.0), (25.0, 26.0), (10.0, 10.0)],
 [(0.0, 0.0), (26.0, 27.0), (10.0, 10.0)]]`

When I convert it to open3d using:
```
pcd = o3d.geometry.PointCloud()
pcd.points = o3d.utility.Vector3dVector(position)
```
I have this error:
`RuntimeError: Unable to cast Python instance to C++ type (compile in debug mode for details)
`
please how can I solve this error?
 Thanks in advance.

**Comment by reptinias** [comment]
Hello there.

The reason you get the error is because you use a python list as an argument to the Vector3dVector.
Converting the list to a numpy array would solve this error.

However when I run the script after conversion, you would get another error, which is due to the structure of the list.
If you restructured the list to look like this:
![image](https://user-images.githubusercontent.com/32259405/172179396-e27e59d8-8ee1-4942-94ff-61c2e22bf2db.png)
You would not get the error.

I have tried to write some code, as example to how you could use the Vector3dVector

`
import open3d as o3d
import numpy as np

array = np.random.random_sample(size = (15,3))

print(array)
position = array.tolist()


listToNp = np.asarray(position)

pcd = o3d.geometry.PointCloud()
pcd.points = o3d.utility.Vector3dVector(listToNp)

o3d.visualization.draw_geometries([pcd])
`

This would result in the output point cloud below:
![image](https://user-images.githubusercontent.com/32259405/172178513-0983535d-5347-4935-b2e8-09fd8b13b93a.png)

Hope this helps you!

**Reply by assia855** [reply]
Thank you so much for your replay. Actually my matrix is in shape `(4089, 3, 2)` and it is already in list. Even while using your code I still have the same problem. Any other idea to make it work. Thanks in advance.
![Capture](https://user-images.githubusercontent.com/62453226/172182170-35e3430d-1a4a-4059-973a-ed7589ad55a5.PNG)

**Reply by reptinias** [reply]
When looking at the documentation the Vector3dVector, take a (n, 3) numpy array. So i unfortunatly don't think it is possible to use that matrix. 
However if you take forexample the first row [(0.0, 0.0), (1.0,  2.0), (14.0. 14.0)] and turn that into [0, 1, 14] it would work. This has to be done for all rows in the matrix.
[Documentation](http://www.open3d.org/docs/release/python_api/open3d.utility.Vector3dVector.html)

**Reply by assia855** [reply]
I'm trying to visualize a graph and this matrix contain the coordinate of the edges, so I can't seem to change it. I already tried to reshape the matrix from (4089, 3, 2) to (4089, 3) but the  output wasn't the same that I 'm expecting.

**Reply by reptinias** [reply]
I'm sorry but I'm not sure how to do this.
My best bet would be to run the matrix through a for loop and add the points one by one to the point cloud.
I'm sorry that i can't help more than that

**Reply by assia855** [reply]
Thanks alot
